src/main/java/backend/Logic.java

```diff
@@ -231,9 +232,24 @@ public void removeUnusedModels(Set<String> reposInUse) {
                 .forEach(models::removeRepoModelById);
     }
 
-    public ImmutablePair<Integer, Long> updateRemainingRate(ImmutablePair<Integer, Long> rateLimits) {
-        uiManager.updateRateLimits(rateLimits);
-        return rateLimits;
+    /**
+     * Updates the GitHub API quota info in the GUI.
+     * @param apiQuotaInfo The GitHub API quota information.
+     * @return the apiQuotaInfo instance
+     */
+    public ApiQuotaInfo updateApiQuotaInfoInGui(ApiQuotaInfo apiQuotaInfo) {
+        uiManager.updateApiQuotaInfo(apiQuotaInfo);
+        return apiQuotaInfo;
```

Comment:
I think thenCompose needs to work together with thenApply? Or we can remove the thenCompose method and use thenRun, but I feel it will make messier?

Sent from Outlook<https://aka.ms/liysn4> on Android



On Sat, Apr 30, 2016 at 8:48 PM -0700, "Nguyen Quoc Dat" <notifications@github.com<mailto:notifications@github.com>> wrote:


In src/main/java/backend/Logic.java<https://github.com/HubTurbo/HubTurbo/pull/1376#discussion_r61678754>:

> @@ -231,9 +232,24 @@ public void removeUnusedModels(Set<String> reposInUse) {
>                  .forEach(models::removeRepoModelById);
>      }
>
> -    public ImmutablePair<Integer, Long> updateRemainingRate(ImmutablePair<Integer, Long> rateLimits) {
> -        uiManager.updateRateLimits(rateLimits);
> -        return rateLimits;
> +    /**
> +     * Updates the GitHub API quota info in the GUI.
> +     * @param apiQuotaInfo The GitHub API quota information.
> +     * @return the apiQuotaInfo instance
> +     */
> +    public ApiQuotaInfo updateApiQuotaInfoInGui(ApiQuotaInfo apiQuotaInfo) {
> +        uiManager.updateApiQuotaInfo(apiQuotaInfo);
> +        return apiQuotaInfo;


Why not use thenRun then?

-
You are receiving this because you were mentioned.
Reply to this email directly or view it on GitHub<https://github.com/HubTurbo/HubTurbo/pull/1376/files/40304ebf9673209f54186e6e7e9e9b5ac02a979e#r61678754>